tests/sentry/projects/project_rules/test_creator.py

```diff
@@ -150,9 +148,14 @@ def test_dual_create_workflow_engine(self):
         assert action.type == Action.Type.PLUGIN
 
     @with_feature("organizations:workflow-engine-issue-alert-dual-write")
-    @patch("sentry.projects.project_rules.creator.IssueAlertMigrator.run")
-    def test_dual_create_workflow_engine__cant_acquire_lock(self, mock_run):
-        mock_run.side_effect = UnableToAcquireLock
+    def test_dual_create_workflow_engine__cant_acquire_lock(self):
+        lock = locks.get(
+            f"workflow-engine-project-error-detector:{self.project.id}",
```

Comment:
It'd probably be helpful to generate lock names in a helper function, in case we need these elsewhere

Reply:
if i see the race condition elsewhere, i'll add a function, but i think the API is the only place where we get spammed